src/actions/submission.js

```diff
@@ -31,8 +59,9 @@ export const fetchSubmission = id => async (dispatch, getStore, { graphqlRequest
     if (errors) {
       // @TODO handle better
       logError(errors);
-    } else if (data.submission) {
-      dispatch(getSubmissionSuccess(data.submission));
+    } else if (data.submission && data.status) {
+      await dispatch(getSubmissionSuccess(data.submission));
+      await dispatch(getSubmissionStatusSuccess(id, data.status));
```

Comment:
Yes, these don't need the "await" and are perfectly safe to dispatch at the same time. I split them up into two actions because I wanted to reuse the latter for after you cast a vote.